### PR TITLE
Include client-gen command line arguments in the output files

### LINKED ISSUE #22001: client-gen: generated files should show command used to generate them
This would be useful if people want to regenerate the clientset.
cc @krousey

### PR DESCRIPTION
**Pull request by caesarxuchao**
Fix #22001
@krousey

### CODE REVIEW

File: cmd/libs/go2idl/client-gen/main.go [hunk 1]
```diff
@@ -74,6 +74,14 @@ func parseInputVersions() (paths []string, groupVersions []unversioned.GroupVers
 func main() {
 	arguments := args.Default()
 	flag.Parse()
+	var cmdArgs string
+	flag.VisitAll(func(f *flag.Flag) {
+		if !f.Changed || f.Name == "verify-only" {
```
Comment (krousey) [inline cmd/libs/go2idl/client-gen/main.go]: What if the defaults change?

Comment (caesarxuchao) [inline cmd/libs/go2idl/client-gen/main.go]: Then the hack/verify-codegen.sh will report the file needs to be regenerated.

**Comment by jianhuiz** [comment]
It is possible to put the command line arguments to separate files in
`pkg/client/clientset_generated` and `pkg/client/typed/generated`
so changing of the arguments (add new api group for example) would not have to affect most of the generated files with just one line comment change

**Comment by caesarxuchao** [comment]
@jianhuiz thanks, how about moving the line to doc.go which comes with every package?

**Comment by jianhuiz** [comment]
@caesarxuchao Thanks, it would be perfect

**Pull request merged**